internal/pkg/flink/internal/store/store.go

```diff
@@ -178,10 +191,36 @@ func (s *Store) waitForPendingStatement(ctx context.Context, statementName strin
 
 	var errorsMsg string
 	if len(capturedErrors) > 0 {
-		errorsMsg = fmt.Sprintf(" Captured retriable errors: %s", strings.Join(capturedErrors, "; "))
+		errorsMsg = fmt.Sprintf("Captured retriable errors: %s", strings.Join(capturedErrors, "; "))
+	}
+
+	return nil, &types.StatementError{
+		Msg: fmt.Sprintf("Statement is still pending after %f seconds.\n If you want to increase the timeout for the client, you can run \"SET table.results-timeout=1200;\" to adjust the maximum timeout in seconds.",
+			timeout.Seconds()),
+		FailureMessage: errorsMsg,
+	}
+}
+
+func (s *Store) getStatusDetail(statementObj flinkgatewayv1alpha1.SqlV1alpha1Statement) string {
+	status := statementObj.GetStatus()
+	phase := types.PHASE(status.GetPhase())
+	if phase != types.FAILED && phase != types.FAILING {
+		return status.GetDetail()
+	}
+
+	if status.GetDetail() != "" {
+		return status.GetDetail()
+	}
+
+	// if the statement is in FAILED or FAILING phase and the status detail field is empty we show the latest exception instead
+	exceptionsResponse, _ := s.client.GetExceptions(s.appOptions.GetEnvironmentId(), statementObj.Spec.GetStatementName(), s.appOptions.GetOrgResourceId())
```

Comment:
Why are we ignoring the error?

Reply:
Because we don't really need to check it here. When an error happens `exceptionsResponse.GetData()` will still return an empty array and thus trigger the `if` which returns an empty string. 
It works the same as this:
`if err != nil || len(exceptionsResponse.GetData()) < 1`

Reply:
Okay makes sense, will handle it explicitly!